csbdeep/io/modelzoo.py

```diff
@@ -0,0 +1,138 @@
+import os
+from datetime import datetime
+from hashlib import sha256
+from zipfile import ZipFile
+
+from ruamel.yaml import YAML
+from tifffile import imsave
+
+from csbdeep.utils.six import Path
+from csbdeep.utils.tf import export_SavedModel
+
+
+class Preprocessing(dict):
+    def __init__(self):
+        pass
+
+
+class Postprocessing(dict):
+    def __init__(self):
+        pass
+
+
+class ZeroMeanUnitVariance(Preprocessing):
+    def __init__(self, mode, axes, mean, std):
+        self["name"] = "zero_mean_unit_variance"
+        self["kwargs"] = {}
+        self["kwargs"]["mode"] = mode
+        self["kwargs"]["axes"] = axes
+        if (mode == "fixed"):
+            self["kwargs"]["mean"] = mean
+            self["kwargs"]["std"] = std
+
+
+class ScaleLinear(Postprocessing):
+    def __init__(self, gain, axes, offset):
+        self["name"] = "scale_linear"
+        self["kwargs"] = {}
+        self["kwargs"]["gain"] = gain
+        self["kwargs"]["offset"] = offset
+        self["kwargs"]["axes"] = axes
+
+
+class ModelZooInput(dict):
+
+    def __init__(self, name, axes, data_type, data_range, halo, min, step,
+                 preprocessing):
+        self["name"] = name
+        self["axes"] = axes
+        self["data_type"] = data_type
+        self["data_range"] = data_range
+        self["halo"] = halo
+        self["shape"] = {}
+        self["shape"]["min"] = min
+        self["shape"]["step"] = step
+        self["preprocessing"] = preprocessing
+
+
+class ModelZooOutput(dict):
+    def __init__(self, name, axes, offset, scale, reference_input, postprocessing):
+        self["name"] = name
+        self["axes"] = axes
+        self["shape"] = {}
+        self["shape"]["offset"] = offset
+        self["shape"]["scale"] = scale
+        self["shape"]["reference_input"] = reference_input
+        self["postprocessing"] = postprocessing
+
+
+class ModelZooWeight(dict):
+    def __init__(self, weight_format, source, modelfile_name, authors, tag, tensorflowversion):
+        self[weight_format] = {}
+        self[weight_format]["authors"] = authors
+        self[weight_format]["source"] = source
+        with open(modelfile_name, "rb") as f:
+            bytes = f.read()
+            self[weight_format]["sha256"] = sha256(bytes).hexdigest()
+        self[weight_format]["tag"] = tag
+        self[weight_format]["tensorflowversion"] = tensorflowversion
+
+
+class ModelZooBaseData(dict):
```

Comment:
Is the file copied from somewhere?

Reply:
Unfortunately not. I'll ask Wei and CC you if we already have something for all of that in place somewhere as a library.

Reply:
No problem, I was just curious.